File: programming_2nd_sem/Triangle.cpp
#include <iostream>
#include <cmath>
using namespace std;

class Triangle {
public:
    Triangle(int sideA, int sideB, int sideC);
    Triangle();
    int perimeter() {return a+b+c;}
    int area(int p) {return sqrt(p*(p-a)*(p-b)*(p-c));}
    int addToSmall();
    void extract() {a-=1; b-=1; c-=1;}
    void print();
    bool rightTrian();
private:
    int a,b,c;
};

Triangle::Triangle(int sideA, int sideB, int sideC){
    a = sideA;
    b = sideB;
    c = sideC;
}

Triangle::Triangle(){
    a = 1;
    b = 1;
    c = 1;
}

int Triangle :: addToSmall(){
    int tmp;
    (a<b) ? tmp=a : tmp=b;
    (tmp<c) ? tmp=tmp : tmp=c;
    return tmp + 1;
}

void Triangle::print(){
    cout<<"A side: "<<a<<endl
        <<"B side: "<<b<<endl
        <<"C side: "<<c<<endl;
}

bool Triangle::rightTrian(){
    int c1 = pow(c,2);
    if(c1 = pow(a,2)+pow(b,2)) return true;
    else return false;
}

int main() {
    Triangle triangle(5,3,4);
    int p = triangle.perimeter();
    cout<<"Perimeter = "<<p<<endl;
    int a = triangle.area(p);
    cout<<"Area = "<<a<<endl;
    triangle.print();
    triangle.extract();
    cout<<"New values: "<<endl;
    triangle.print();

    return 0;
}

File: 4th_JAVA/OOP_Classes_and_objects/src/com/company/Triangle.java
package com.company;

public class Triangle {
    private double side1;
    private double side2;
    private double side3;

    public Triangle() {
        this.side1 = 100;
        this.side2 = 100;
        this.side3 = 100;
    }

    public Triangle (double side){
        this.side1 = this.side2 = this.side3 = side;
    }

    public Triangle(double side1, double side2, double side3) {
        this.side1 = side1;
        this.side2 = side2;
        this.side3 = side3;
    }

    @Override
    public String toString() {
        return "Triangle{" +
                "side1=" + side1 +
                ", side2=" + side2 +
                ", side3=" + side3 +
                '}';
    }

    public boolean isTriangle(){
        return (side1 + side2 > side3) && (side1 + side3 > side2) && (side2 + side3 > side1);
    }

    public double perimeter(){
        return side1+side2+side3;
    }

    public double area(){
        return Math.sqrt(perimeter()/2 * (perimeter()/2 - side1) * (perimeter()/2 - side2) * (perimeter()/2 - side3));
    }

    public void increaseSideLength(double num){
        if(num < 0){
            System.out.println("Not valid");
        }
        else{
            this.side1 += num;
            this.side2 += num;
            this.side3 += num;
        }
    }

    public boolean isEquilateral(){
        return side1 == side2 && side1 == side3;
    }

    public boolean isRectangular(){
        boolean isRect = (Math.pow(side1, 2) == Math.pow(side2, 2) + Math.pow(side3, 2)) || (Math.pow(side2, 2) == Math.pow(side1, 2) + Math.pow(side3, 2)) || (Math.pow(side3, 2) == Math.pow(side2, 2) + Math.pow(side1, 2));
        return isRect;
    }
}
